### ISSUE TITLE
train_x_vectors.yaml's output doesnot have the file of "mean_var_norm_emb.ckpt"

### ISSUE THREAD

**Issue by zyliutju** [speechbrain/speechbrain#678]
hi
The ouput of `python train_speaker_embeddings.py hyperparams/train_x_vectors.yaml` doesnot have the file of "mean_var_norm_emb.ckpt", which will be used in "verification_plda_xvector.yaml"; is it（"mean_var_norm_emb.ckpt"） necessary  for "verification_plda_xvector.yaml"？，thank you.

**Comment by mravanelli** [comment]
I think you can have very similar performance if you comment in the yaml file this:

```
#mean_var_norm_emb: !ref <mean_var_norm_emb>

#mean_var_norm_emb: !ref <pretrain_path>/mean_var_norm_emb.ckpt
```
This way the normalization will be done on-the-fly, but this is probably fine. 
Note also that we now have better performance with the ECAPA-TDNN model.

**Comment by zyliutju** [comment]
Thank you for your reply. I am a beginner in Speaker recognition，and it is my first time using this toolkit。
I used Vox2 for traning xvector，using data augment in \recipes\VoxCeleb\SpeakerRec\hparams\train_x_vectors.yaml
My experience using \recipes\VoxCeleb\SpeakerRec\hparams\verification_plda_xvector.yaml performance is：
EER(%)=3.488341
min_dcf=0.407939

\recipes\VoxCeleb\SpeakerRec\hparams\verification_plda_xvector.yaml need "mean_var_norm_emb.ckpt",but during train phrase, there hasnot this file, so I delete #mean_var_norm_emb: !ref <pretrain_path>/mean_var_norm_emb.ckpt，and get this performance .

My quesiton is：
in the \recipes\VoxCeleb\SpeakerRec\hparams folder，the train_x_vectors.yaml only has the 
“mean_var_norm: !new:speechbrain.processing.features.InputNormalization
    norm_type: sentence
    std_norm: False”，and the “train_speaker_embeddings.py” only call the model “mean_var_norm”.
but when verify phrase, "verification_plda_xvector.yaml" need "mean_var_norm_emb.ckpt", so how can I get the output of this "mean_var_norm_emb.ckpt" file? I think the output of 
"mean_var_norm_emb: !new:speechbrain.processing.features.InputNormalization
    norm_type: global
    std_norm: False"  can do this. But I cannot find where it's called in the whole project。

**Comment by zyliutju** [comment]
> I think you can have very similar performance if you comment in the yaml file this:
> 
> ```
> #mean_var_norm_emb: !ref <mean_var_norm_emb>
> 
> #mean_var_norm_emb: !ref <pretrain_path>/mean_var_norm_emb.ckpt
> ```
> 
> This way the normalization will be done on-the-fly, but this is probably fine.
> Note also that we now have better performance with the ECAPA-TDNN model.

Thank you for your reply. I am a beginner in Speaker recognition，and it is my first time using this toolkit。
I used Vox2 for traning xvector，using data augment in \recipes\VoxCeleb\SpeakerRec\hparams\train_x_vectors.yaml
My experience using \recipes\VoxCeleb\SpeakerRec\hparams\verification_plda_xvector.yaml performance is：
EER(%)=3.488341
min_dcf=0.407939

\recipes\VoxCeleb\SpeakerRec\hparams\verification_plda_xvector.yaml need "mean_var_norm_emb.ckpt",but during train phrase, there hasnot this file, so I delete #mean_var_norm_emb: !ref <pretrain_path>/mean_var_norm_emb.ckpt，and get this performance .

My quesiton is：
in the \recipes\VoxCeleb\SpeakerRec\hparams folder，the train_x_vectors.yaml only has the
“mean_var_norm: !new:speechbrain.processing.features.InputNormalization
norm_type: sentence
std_norm: False”，and the “train_speaker_embeddings.py” only call the model “mean_var_norm”.
but when verify phrase, "verification_plda_xvector.yaml" need "mean_var_norm_emb.ckpt", so how can I get the output of this "mean_var_norm_emb.ckpt" file? I think the output of
"mean_var_norm_emb: !new:speechbrain.processing.features.InputNormalization
norm_type: global
std_norm: False" can do this. But I cannot find where it's called in the whole project。

**Comment by mravanelli** [comment]
I think the best approach here is to store the mean_var_norm_emb.ckpt when training withing train_speaker_embeddings.py (where it is not actually used but make things simpler). Let me do the modification as soon as I can.

**Comment by zyliutju** [comment]
thank you.

**Comment by rohaantahir** [comment]
Hi could you please tell me how mean_var_norm_emb.ckpt while custom training using Voxceleb Recipe for speaker verification.
